Assets/HoloToolkit/Input/Scripts/KeywordManager.cs

```diff
@@ -2,79 +2,97 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System.Collections.Generic;
-using System.Linq;
 using UnityEngine;
-using UnityEngine.Events;
 using UnityEngine.Windows.Speech;
 
 namespace HoloToolkit.Unity
 {
     /// <summary>
-    /// KeywordManager allows you to specify keywords and methods in the Unity
+    /// KeywordManager allows you to specify keywords in the Unity
     /// Inspector, instead of registering them explicitly in code.
     /// This also includes a setting to either automatically start the
     /// keyword recognizer or allow your code to start it.
     ///
     /// IMPORTANT: Please make sure to add the microphone capability in your app, in Unity under
     /// Edit -> Project Settings -> Player -> Settings for Windows Store -> Publishing Settings -> Capabilities
     /// or in your Visual Studio Package.appxmanifest capabilities.
+    /// 
+    /// Please, also make sure to also set your keyword responses in the Inspector window under Keywords and Responses Array.
     /// </summary>
-    public partial class KeywordManager : MonoBehaviour
+    public partial class KeywordManager : Singleton<KeywordManager>
```

Comment:
KeywordRecognizer has been optimized to have many of them around the scene, to facilitate turning keywords on/off when they're valid/invalid. I'm happy to rename this script if you think Manager implies only one. Its purpose is to manage setting up a keyword recognizer and make it easier by setting keywords in the Inspector, taking the code setup out of the hands of the developer.

What if I have a KeywordManager I want to turn on when I gaze at a specific GameObject? How does having this as a Singleton handle turning on and off specific keywords based on context? In the current implementation, we can place a KeywordManager on the object and turn it on in OnGazeEnter.